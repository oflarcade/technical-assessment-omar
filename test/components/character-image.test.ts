import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import CharacterImage from '@/components/rick-and-morty/character-image.vue';


const character = {
    id: 1,
    name: 'Rick Sanchez',
    species: 'Human',
    image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
};

describe('RickAndMortyCharacterImage.vue', () => {
    it('renders correctly with given character image', () => {
        const wrapper = mount(CharacterImage, {
            props: {
                character,
            },
        });

        const img = wrapper.find('img');
        expect(img.exists()).toBe(true);
        expect(img.attributes('src')).toBe(character.image);
    });

    it('matches snapshot', () => {
        const wrapper = mount(CharacterImage, {
            props: {
                character,
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });
});
