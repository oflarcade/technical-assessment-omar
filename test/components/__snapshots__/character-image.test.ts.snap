// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RickAndMortyCharacterImage.vue > matches snapshot 1`] = `"<img data-v-594c78aa="" data-nuxt-img="" class="image" src="https://rickandmortyapi.com/api/character/avatar/1.jpeg">"`;
